stanford-sw/test/src/edu/stanford/ClosedLibrariesTests.java

```diff
@@ -0,0 +1,129 @@
+		solrFldMapTest.assertSolrFldValue(rec, fldName, "Green");
+
+	    // Test case #3
+		rec = factory.newRecord();
+	    rec.setLeader(factory.newLeader("04473caa a2200313Ia 4500"));
+		cf = factory.newControlField("001", "aactualRecord");
+		rec.addVariableField(cf);
+		df999 = factory.newDataField("999", '0', '0');
+		df999.addSubfield(factory.newSubfield('a', "QB1 .A8133 V.402"));
+		df999.addSubfield(factory.newSubfield('w', "LC"));
+		df999.addSubfield(factory.newSubfield('c', "1"));
+		df999.addSubfield(factory.newSubfield('i', "36105210713652"));
+		df999.addSubfield(factory.newSubfield('l', "POP-SCI"));
+		df999.addSubfield(factory.newSubfield('m', "SCIENCE"));
+		df999.addSubfield(factory.newSubfield('t', "STKS"));
+		rec.addVariableField(df999);
+
+		df999 = factory.newDataField("999", '0', '0');
+		df999.addSubfield(factory.newSubfield('a', "DO1 .A8133 V.402"));
+		df999.addSubfield(factory.newSubfield('w', "LC"));
+		df999.addSubfield(factory.newSubfield('c', "1"));
+		df999.addSubfield(factory.newSubfield('i', "36105210713654"));
+		df999.addSubfield(factory.newSubfield('k', "STACKS"));
+		df999.addSubfield(factory.newSubfield('m', "CHEMCHMENG"));
+    	df999.addSubfield(factory.newSubfield('t', "STKS"));
+    	rec.addVariableField(df999);
+
+    	// Index but no building_facet value
+		// solrFldMapTest.assertNoSolrFld(rec,fldName);
+
+    // Building_facet value is Science (Li and Ma)
+		solrFldMapTest.assertSolrFldValue(rec, fldName, "Science (Li and Ma)");
+
+		// Ignore item with library CHEMCHMENG
+		solrFldMapTest.assertSolrFldHasNoValue(rec, "barcode", "36105210713654");
+	}
+
+	/** In order to check the exclusion of closed library items without open library items, a fresh index needs to be created so
+	 *   these tests require a test file
+	 */
+@Test
+	public void testClosedLibrariesNotIndexed() throws ParserConfigurationException, IOException, SAXException, SolrServerException
+	{
+		// index test records
+		createFreshIx(testDataFname);
+
+		// Test Case #1 Make sure record with BIOLOGY only does not get indexed
+		assertZeroResults("id", "doNotIndexBio");
+
+		// Test Case #1 Make sure record with CHEMCHMENG only does not get indexed
+		assertZeroResults("id", "doNotIndexChem");
+
+    // Test Case #1 Make sure record with MATH-CS only does not get indexed
+    assertZeroResults("id", "doNotIndexMath");
+
+		// Test Case #2 Make sure record with GREEN gets indexed
+		assertResultSize("id", "noClosedLib", 1);
+
+    // Test Case #3 Make sure record with multiple libraries including BIOLOGY and CHEMCHMENG in 999$m gets indexed
+      // but no building_facet for Biology or Chemistry/Chemical Eng
+    assertResultSize("id", "indexMultL", 1);
+
+		// Make sure no records have a building_facet belonging to a closed library
+		assertZeroResults("building_facet", "Biology (Falconer)");
+    assertZeroResults("building_facet", "Chemistry & ChemEng (Swain)");
+    assertZeroResults("building_facet", "Math & Statistics");
+	}
+}
```

Comment:
in a perfect world, you might consider separate test methods for the separate test cases:  #1, #2, #3, the facet ...